src/python/pants/backend/project_info/tasks/ide_gen.py

```diff
@@ -35,6 +36,11 @@
 from pants.util.dirutil import safe_mkdir, safe_walk
 
 
+deprecated_module('1.7.0.dev0',
```

Comment:
Does mean `class EnsimeGen(IdeGen)` is auto deprecated as well?

Reply:
Yes, any subclass must import ide_gen.py, which triggers the deprecation warning.